controllers/tykoasapidefinition_controller.go

```diff
@@ -140,9 +140,15 @@ func (r *TykOasApiDefinitionReconciler) createOrUpdateTykOASAPI(ctx context.Cont
 ) {
 	var cm v1.ConfigMap
 
+	confMapNs := tykOASCrd.Spec.TykOAS.ConfigmapRef.Namespace
+
 	ns := types.NamespacedName{
-		Name:      tykOASCrd.Spec.TykOAS.ConfigmapRef.Name,
-		Namespace: tykOASCrd.Spec.TykOAS.ConfigmapRef.Namespace,
+		Name: tykOASCrd.Spec.TykOAS.ConfigmapRef.Name,
+	}
+
+	ns.Namespace = confMapNs
+	if confMapNs == "" {
+		ns.Namespace = tykOASCrd.Namespace
 	}
```

Comment:
This looks good to me. But I guess we have redundant operations on `ns` object while setting the `Namespace`.
So, in my opinion we can simplify even more this code logic, as follows:

```go
	configMapNs := tykOASCrd.Spec.TykOAS.ConfigmapRef.Namespace
	if configMapNs == "" {
		configMapNs = tykOASCrd.Namespace
	}

	ns := types.NamespacedName{
		Name:      tykOASCrd.Spec.TykOAS.ConfigmapRef.Name,
		Namespace: configMapNs,
	}
```
What do you think?